src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,58 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
+
+        public static ImmutableArray<TypeWithAnnotations> CreateTypeArguments(ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            return typeParameters.SelectAsArray(_ => s_instance);
```

Comment:
>SelectAsArray [](start = 34, length = 13)

Can we add a Repeate helper instead? All we are using from ```typeParameters``` is length.

Reply:
We could, but I changed the code from populating an ArrayBuilder to using `SelectAsArray` in previous feedback. I don't think this is worth changing back again.

---
In reply to: [503346990](https://github.com/dotnet/roslyn/pull/48270#discussion_r503346990) [](ancestors = 503346990)

Reply:
> I changed the code from populating an ArrayBuilder to using SelectAsArray in previous feedback. I don't think this is worth changing back again.

This isn't exactly what I am suggesting.


---
In reply to: [503602627](https://github.com/dotnet/roslyn/pull/48270#discussion_r503602627) [](ancestors = 503602627,503346990)